r/R/dplyr-funcs-datetime.R

```diff
@@ -189,6 +189,65 @@ register_bindings_datetime <- function() {
   })
 }
 
+register_bindings_duration <- function() {
+  register_binding("difftime", function(time1,
+                                        time2,
+                                        tz,
+                                        units = c("auto", "secs", "mins",
+                                                  "hours", "days", "weeks")) {
+    units <- match.arg(units)
+    if (units != "secs") {
+      abort("`difftime()` with units other than seconds not supported in Arrow")
+    }
+
+    if (!missing(tz)) {
+      warn("`tz` is an optional argument to `difftime()` in R and will not be used in Arrow")
+    }
+
+    time1 <- build_expr("cast", time1, options = cast_options(to_type = timestamp()))
+    time2 <- build_expr("cast", time2, options = cast_options(to_type = timestamp()))
```

Comment:
I'm not sure if this is really necessary, since casting to the same type should be a no-op, but you might want to confirm that it is and if it does add overhead wrap these in `if (!is.timestamp(time1)) { cast }`

Reply:
I was using this step to also go from, for example, a POSIXct R object to an `Expression` and I thought it might be cleaner than having a bunch of `if (!(inherits("Expression", x) && x$type_id() %in% ...))`

Reply:
I think this is done.